src/data/matrix/hadamard.lean

```diff
@@ -0,0 +1,142 @@
+/-
+Copyright (c) 2021 Lu-Ming Zhang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Lu-Ming Zhang
+-/
+import data.matrix.notation
+import linear_algebra.matrix.trace
+import data.complex.basic
+
+/-!
+# Hadamard product of matrices
+
+This file defines the Hadamard product `matrix.Hadamard`
+and contains basic properties about them.
+
+## Main definition
+
+- `matrix.Hadamard`: defines the Hadamard product,
+  which is the pointwise product of two matrices of the same size.
+
+## Notation
+
+* `⊙`: the Hadamard product `matrix.Hadamard`;
+
+## References
+
+*  <https://en.wikipedia.org/wiki/Hadamard_product_(matrices)>
+
+## Tags
+
+Hadamard product, Hadamard
+-/
+
+variables {α β γ I J K L M N: Type*}
+variables {R : Type*}
+variables {m n p q r s t: ℕ}
```

Comment:
As commented in the other PR, convention is to use `l m n p` or similar for matrix indices, not `I J K L`